marketplace-builder/src/builder.rs

```diff
@@ -202,31 +203,64 @@ impl BuilderConfig {
         let events_url = hotshot_events_api_url.clone();
         tracing::info!("Running permissionless builder against hotshot events API at {events_url}",);
 
-        let hooks = hooks::EspressoHooks {
-            namespaces: namespaces.into_iter().collect(),
-            solver_api_url,
-            builder_api_base_url: hotshot_builder_apis_url.clone(),
-            bid_key_pair: builder_key_pair,
-            bid_amount,
-        };
-
-        async_spawn(async move {
-            let res = run_non_permissioned_standalone_builder_service(
-                hooks,
-                BroadcastSenders {
-                    transactions: tx_sender,
-                    da_proposal: da_sender,
-                    quorum_proposal: qc_sender,
-                    decide: decide_sender,
-                },
-                events_url,
-            )
-            .await;
-            tracing::error!(?res, "builder service exited");
-            if res.is_err() {
-                panic!("Builder should restart.");
-            }
-        });
+        if fallback {
```

Comment:
I don't think marketplace-builder-core will behave correctly if we spawn two `BuilderState`'s for the same chain.
Can we unify hooks as well and put a flag for fallback there?